src/main/java/org/opensearch/geospatial/action/upload/UploadStats.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.action.upload;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+
+/**
+ * Contains the total upload stats
+ */
+public final class UploadStats implements ToXContent {
+
+    public static final String ROOT_FIELD = "upload";
+
+    public enum FIELDS {
+        total,
+        metrics
+    }
+
+    private static UploadStats instance = new UploadStats();
+
+    private final List<UploadMetric> metrics;
+    private final CounterMetric totalAPICount;
+
+    /**
+     * @return Singleton instance of UploadStats
+     */
+    public static UploadStats getInstance() {
+        return instance;
+    }
+
+    UploadStats() {
+        metrics = new ArrayList<>();
+        totalAPICount = new CounterMetric();
+    }
+
+    /**
+     * Add new metric to {@link UploadStats}
+     * @param newMetric
+     */
+    public void addMetric(UploadMetric newMetric) {
+        Objects.requireNonNull(newMetric, "metric cannot be null");
+        metrics.add(newMetric);
```

Comment:
Do we check for duplicate metrics anywhere?

Reply:
changed to HashSet and using metric ID to validate duplicate metrics